### ISSUE TITLE
[data request] NYU Depth Dataset V2

### ISSUE THREAD

**Issue by lgeiger** [tensorflow/datasets#838]
* Name of dataset: NYU Depth Dataset V2
* URL of dataset: https://cs.nyu.edu/~silberman/datasets/nyu_depth_v2.html
* License of dataset: unknown
* Short description of dataset and use case(s):
  The NYU-Depth V2 data set is comprised of video sequences from a variety of indoor scenes as recorded by both the RGB and Depth cameras from the Microsoft Kinect. It features:
  - 1449 densely labeled pairs of aligned RGB and depth images
  - 464 new scenes taken from 3 cities
  - 407,024 new unlabeled frames
  - Each object is labeled with a class and an instance number (cup1, cup2, cup3, etc)

This would be a great addition to `tensorflow/datasets` as it currently doesn't include datasets for depth estimation tasks.

Folks who would also like to see this dataset in `tensorflow/datasets`, please thumbs-up so the developers can know which requests to prioritize.

And if you'd like to contribute the dataset (thank you!), see our [guide to adding a dataset](https://github.com/tensorflow/datasets/blob/master/docs/add_dataset.md).

**Comment by f1recracker** [comment]
I'd be happy to take this if no one is working on this.

**Comment by lgeiger** [comment]
I have implemented a version of this dataset [here](https://github.com/lgeiger/datasets/tree/nyu-depth-v2) which requires #892.
It is based on the dataset published at https://github.com/ialhashim/DenseDepth/#data which includes inpainted depth maps for parts of the raw dataset as well but is not complete. I might clean this up and upstream it next month but feel free to take it over if you have time.

**Comment by hermannsblum** [comment]
@lgeiger Do you plan on creating a PR with your version as #892 is now merged?

**Comment by lgeiger** [comment]
> @lgeiger Do you plan on creating a PR with your version as #892 is now merged?

I don't have current plans to upstream it since the raw dataset is not complete, but it might be useful anyway. I opened https://github.com/ialhashim/DenseDepth/issues/90 to check if we are allowed to use this URL.

**Comment by Conchylicultor** [comment]
Why not creating the dataset from the original website ? https://cs.nyu.edu/~silberman/datasets/nyu_depth_v2.html

It provides links to the full data and annotations. I think this is what users would expect when using this dataset, no ?

**Comment by adiagarwalgoogle** [comment]
This issue is being closed as NYU Depth Dataset V2 is now part of TensorFlow datasets package.
Please refer to https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/image/nyu_depth_v2.py and https://www.tensorflow.org/datasets/catalog/nyu_depth_v2 for more info.
Thank you for your contribution

**Comment by debashishpal81** [comment]
It is mentioned that NYU v2 dataset is now available as Tensorflow dataset package. However I am unable to download it. This is my simple code to load it as a TF dataset:

#!/usr/bin/python3
import tensorflow as tf
import matplotlib.pyplot as plt
import numpy as np
import tensorflow_datasets as tfds
import os
import sys

def main():
    print('Python       :', sys.version.split('\n')[0])
    print('Numpy        :', np.__version__)
    print('Tensorflow   :', tf.__version__)

    ds = tfds.load('nyu_depth_v2', split='train', shuffle_files=True)
    # ds = tfds.load('kitti', split='train', batch_size=16, shuffle_files=False)
    assert isinstance(ds, tf.data.Dataset)
    print(ds)

if __name__ == "__main__":
    main()
